lang/src/org/partiql/lang/planner/PlannerPipeline.kt

```diff
@@ -40,6 +41,38 @@ import org.partiql.lang.syntax.SqlParser
 import org.partiql.lang.syntax.SyntaxException
 import org.partiql.lang.types.CustomType
 
+/**
+ * Represents a pass over the physical plan that accepts a physical plan and returns a modified
+ * physical plan.
+ *
+ * Passes accept as input a [PartiqlPhysical.Plan] which is cloned & modified in some way before being returned.
+ * A second input to the pass is an instance of [ProblemHandler], which can be used to report semantic errors and
+ * warnings to the query author.
+ *
+ * Examples of passes:
+ *
+ * - Select optimal physical operator implementations.
+ * - Push down predicates or projections.
+ * - Convert filter predicates to index lookups.
+ * - Fold constants.
+ * - And many, many others, some will be specific to the application embedding PartiQL.
+ *
+ * Notes on exceptions and semantic problems:
+ *
+ * - The passes may throw any exception, however these will always abort query planning and bypass the user-friendly
+ * error reporting ([ProblemHandler]) mechanisms used for
+ * [syntax and semantic errors](https://www.educative.io/edpresso/what-is-the-difference-between-syntax-and-semantic-errors))
```

Comment:
nit: extra right paren

```suggestion
 * [syntax and semantic errors](https://www.educative.io/edpresso/what-is-the-difference-between-syntax-and-semantic-errors)
```